### ISSUE TITLE
Wrapping RandomCrop in Augmentation/ImageSequential returns wrong inverse

### ISSUE THREAD

**Issue by miquelmarti** [kornia/kornia#1491]
### Describe the bug

When wrapping even a single RandomCrop augmentation with AugmentationSequential or ImageSequential, the expected inverse is not correctly computed.

### Reproduction steps

RandomCrop inverse is correctly computed:
```
import kornia.augmentation as K
import torch

a = torch.rand((3,3)) * 10
a = a.unsqueeze(dim=0).unsqueeze(dim=0)

t = K.RandomCrop((2,2), cropping_mode="resample", padding=1, padding_mode='constant', fill=0)
b = t(a)

print(a)
print(b)
print(t.inverse(b))
```
```
tensor([[[[2.3337, 7.5095, 6.7896],
          [3.5924, 9.5243, 5.8823],
          [8.7506, 4.6143, 5.2864]]]])
tensor([[[[2.3337, 7.5095],
          [3.5924, 9.5243]]]])
tensor([[[[2.3337, 7.5095, 0.0000],
          [3.5924, 9.5243, 0.0000],
          [0.0000, 0.0000, 0.0000]]]]
```

When wrapping the augmentation with AugmentationSequential or ImageSequential:
```
t2 = K.AugmentationSequential(t)
c = t2(a)

print(a)
print(c)
print(t2.inverse(c))
```
```
tensor([[[[2.3337, 7.5095, 6.7896],
          [3.5924, 9.5243, 5.8823],
          [8.7506, 4.6143, 5.2864]]]])
tensor([[[[2.3337, 7.5095],
          [3.5924, 9.5243]]]])
tensor([[[[2.3337]]]])
```

Using a larger input size the result is the same. Also, this behaviour seems to depend on the padding.
```
a = torch.rand((300,300)) * 10
a = a.unsqueeze(dim=0).unsqueeze(dim=0)
t = K.RandomCrop((200,200), cropping_mode="resample", padding=1, padding_mode='constant', fill=0)
t2 = K.ImageSequential(t)
b = t(a)
print(t.inverse(b).shape)
c = t2(a)
print(t2.inverse(c).shape)
```
```
torch.Size([1, 1, 300, 300])
torch.Size([1, 1, 298, 298])
```
```
a = torch.rand((300,300)) * 10
a = a.unsqueeze(dim=0).unsqueeze(dim=0)
t = K.RandomCrop((200,200), cropping_mode="resample", padding=0, padding_mode='constant', fill=0)
t2 = K.ImageSequential(t)
b = t(a)
print(t.inverse(b).shape)
c = t2(a)
print(t2.inverse(c).shape)
```
```
torch.Size([1, 1, 300, 300])
torch.Size([1, 1, 300, 300])
```
### Expected behavior

Same inverse result as without wrapping in container.

### Environment

```shell
Collecting environment information...
PyTorch version: 1.10.0
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.27

Python version: 3.8.5 (default, Sep  4 2020, 07:30:14)  [GCC 7.3.0] (64-bit runtime)
Python platform: Linux-4.15.0-162-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: 
GPU 0: NVIDIA TITAN X (Pascal)
GPU 1: NVIDIA TITAN X (Pascal)
GPU 2: NVIDIA TITAN Xp

Nvidia driver version: 470.82.00
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.1
[pip3] torch==1.10.0
[pip3] torchvision==0.11.1
[conda] blas                      1.0                         mkl    anaconda
[conda] cudatoolkit               11.3.1               ha36c431_9    conda-forge
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] mkl                       2019.4                      243    anaconda
[conda] mkl-service               2.3.0            py38he904b0f_0    anaconda
[conda] mkl_fft                   1.2.0            py38h23d657b_0    anaconda
[conda] mkl_random                1.1.0            py38h962f231_0    anaconda
[conda] numpy                     1.19.1           py38hbc911f0_0    anaconda
[conda] numpy-base                1.19.1           py38hfa32c7d_0    anaconda
[conda] pytorch                   1.10.0          py3.8_cuda11.3_cudnn8.2.0_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchvision               0.11.1               py38_cu113    pytorch
```

**Comment by shijianjian** [comment]
Can you try use a bigger image? I suppose that is something related to the SVD. It is inaccurate with a small image like 3*3 size.

**Comment by miquelmarti** [comment]
Ok, I tried with 300x300 and the output for the inverse of `RandomCrop((200,200), padding=1, ....)` has size 300x300 as expected, while the wrapped version in AugmentationSequential has size 298x298.

I set `padding=0` suspecting there was some issue around this and indeed the output size of the inverse becomes correct. Not sure what this has to do with SVD.

**Comment by shijianjian** [comment]
> Ok, I tried with 300x300 and the output for the inverse of `RandomCrop((200,200), padding=1, ....)` has size 300x300 as expected, while the wrapped version in AugmentationSequential has size 298x298.

> I set `padding=0` suspecting there was some issue around this and indeed the output size of the inverse becomes correct. 

Okay. Can you update the desciption with the corresponding code snippet accordingly? I will test when I am back.

> Not sure what this has to do with SVD.

Not SVD, I remember it is something about the matrix inversion or something. Just put a random word, sorry. :)
